### PR TITLE
Add `mc-sgx-sync::sys::locks::RwLock`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #14** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/14" target="_blank">Graphite</a> 
    * **PR #13** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/13" target="_blank">Graphite</a> 
      * **PR #15** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/15" target="_blank">Graphite</a> 
        * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/22" target="_blank">Graphite</a> 
          * **PR #24** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/24" target="_blank">Graphite</a> 
            * **PR #30** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/30" target="_blank">Graphite</a> 
              * **PR #31** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/31" target="_blank">Graphite</a> 
                * **PR #32** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/32" target="_blank">Graphite</a> 
                  * **PR #33** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/33" target="_blank">Graphite</a> 
                    * **PR #34** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/34" target="_blank">Graphite</a> 
                      * **PR #35** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/35" target="_blank">Graphite</a> 
                        * **PR #36** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/36" target="_blank">Graphite</a> 
                          * **PR #39** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/39" target="_blank">Graphite</a> 
                            * **PR #40** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/40" target="_blank">Graphite</a> 
                              * **PR #41** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/41" target="_blank">Graphite</a> 
                                * **PR #42** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/42" target="_blank">Graphite</a> 
                                  * **PR #43** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/43" target="_blank">Graphite</a> 
                                    * **PR #44** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/44" target="_blank">Graphite</a>  👈
        * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/21" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/44?utm_source=stack-comment).

File: sync/src/sys/locks/rwlock.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+/// SGX rwlock backend to use with the common Rust std lib
+/// [`RwLock`](https://doc.rust-lang.org/std/sync/struct.RwLock.html) interface.
+///
+///
+/// Read locks are implemented using a reference count. This means that clients
+/// are responsible for managing which threads hold the read locks and ensuring
+/// threads only unlock if they are currently holding a reader lock. Write locks
+/// do track which thread owns them.
+pub(crate) struct RwLock {
```
Comment (varsha888) [inline sync/src/sys/locks/rwlock.rs:17]: Could this be renamed to `ReadWriteLock` ? You can leave it as is If `rw` is intuitive in rust.

**Review by varsha888 — APPROVED** [review]

File: sync/src/sys/locks/rwlock.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply sync/src/sys/locks/rwlock.rs:17]: I think for consistency with https://doc.rust-lang.org/std/sync/struct.RwLock.html and it's backend implementations I'm going to leave this as `RwLock`

**Review by varsha888 — APPROVED** [review]

**Review by NotGyro — APPROVED** [review]
This look good to me.

**Comment by nick-mobilecoin** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: uCoNH97G95Be2subCEb9.
See details on [graphite.dev](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/44)

**Comment by nick-mobilecoin** [comment]
@varsha888 bumping for re-review due to re-stack

**Review by varsha888 — APPROVED** [review]

**Pull request merged**